mllib/src/main/scala/org/apache/spark/ml/regression/DecisionTreeRegressor.scala

```diff
@@ -39,7 +40,7 @@ import org.apache.spark.sql.DataFrame
 @Experimental
 final class DecisionTreeRegressor(override val uid: String)
   extends Predictor[Vector, DecisionTreeRegressor, DecisionTreeRegressionModel]
-  with DecisionTreeParams with TreeRegressorParams {
+  with TreeRegressorParams {
```

Comment:
I think it’s necessary to make this change because of ```validateAndTransformSchema``` should be reimplemented for each *Regressor and *RegressionModel if we want to output variance of prediction. So the best way is to implement ```validateAndTransformSchema``` in ```TreeRegressorParams```, then all *Regressor and *RegressionModel extends it can provide variance of prediction.